cmd/kubeadm/app/cmd/init.go

```diff
@@ -156,6 +156,10 @@ func AddInitConfigFlags(flagSet *flag.FlagSet, cfg *kubeadmapiext.MasterConfigur
 		"Specify range of IP addresses for the pod network; if set, the control plane will automatically allocate CIDRs for every node.",
 	)
 	flagSet.StringVar(
+		&cfg.KubeProxy.BindAddress, "kube-proxy-bind-address", cfg.KubeProxy.BindAddress,
```

Comment:
We have basically declared a freeze on new CLI flags to avoid bloat. Could we extract 0.0.0.0 or ::0 from whether the API server advertise address is IPv6 or not? Does that make sense?

Reply:
@luxas: Cc @rpothier @danehans 
Yes, in fact rpothier has started looking into this already. I had started down that path originally, but it's not obvious how the section of code that's using the bindAddress has any access to the API advertise address, and whether it can tell, by the time we're executing, whether the bindAddress was explicitly set to let's say 0.0.0.0 vs. defaulting to 0.0.0.0. So I thought at the time that adding a knob to kubeadm would be the "path of least resistance". So now we're reconsidering, and this is EXACTLY what rpothier is looking into.

Reply:
@luxas (Cc @rpothier @danehans) If the auto-detect is not feasible without major tear-up and rearrangement of the kube-proxy code, could we consider @danehans suggestion of using the "extra args" approach that kubeadm provides for other Kubernetes control plane components?  (See danehans' comment from 10 days prior).

Reply:
@luxas: i.e. the extra args string would be almost opaque to kubeadm, but we'd do some parsing of the args string(s) and stuff the values into a kube-proxy config file.

Reply:
@leblancd This will be configurable with the config file in any case, just set `.kubeProxy.config.bindAddress`. That will be used instead of an "extra args" string-string map (which has a lot of disadvantages). We're eventually gonna use ComponentConfig for all components instead of those maps